plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -851,18 +855,41 @@ public static Transaction newCreateTableTransaction(TrinoCatalog catalog, Connec
         SortOrder sortOrder = parseSortFields(schema, getSortOrder(tableMetadata.getProperties()));
 
         if (replace) {
-            return catalog.newCreateOrReplaceTableTransaction(session, schemaTableName, schema, partitionSpec, sortOrder, tableLocation, createTableProperties(tableMetadata, allowedExtraProperties));
+            return catalog.newCreateOrReplaceTableTransaction(
+                    session,
+                    schemaTableName,
+                    schema,
+                    partitionSpec,
+                    sortOrder,
+                    tableLocation,
+                    createTableProperties(tableMetadata, allowedExtraProperties, session));
         }
-        return catalog.newCreateTableTransaction(session, schemaTableName, schema, partitionSpec, sortOrder, tableLocation, createTableProperties(tableMetadata, allowedExtraProperties));
+        return catalog.newCreateTableTransaction(
+                session,
+                schemaTableName,
+                schema,
+                partitionSpec,
+                sortOrder,
+                tableLocation,
+                createTableProperties(tableMetadata, allowedExtraProperties, session));
     }
 
-    public static Map<String, String> createTableProperties(ConnectorTableMetadata tableMetadata, Predicate<String> allowedExtraProperties)
+    public static Map<String, String> createTableProperties(ConnectorTableMetadata tableMetadata, Predicate<String> allowedExtraProperties, ConnectorSession session)
     {
         ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builder();
         IcebergFileFormat fileFormat = IcebergTableProperties.getFileFormat(tableMetadata.getProperties());
         propertiesBuilder.put(DEFAULT_FILE_FORMAT, fileFormat.toIceberg().toString());
         propertiesBuilder.put(FORMAT_VERSION, Integer.toString(IcebergTableProperties.getFormatVersion(tableMetadata.getProperties())));
-        propertiesBuilder.put(COMMIT_NUM_RETRIES, Integer.toString(IcebergTableProperties.getMaxComiRetry(tableMetadata.getProperties())));
+        propertiesBuilder.put(COMMIT_NUM_RETRIES, Integer.toString(IcebergTableProperties.getMaxCommitRetry(tableMetadata.getProperties())));
+
+        switch (fileFormat) {
+            case PARQUET -> getCompressionCodec(session).getParquetCompressionCodec()
```

Comment:
Hi @ebyhr, do you think we should introduce 3 matching properties for `write.[parquet|avro|orc].compression-codec` in  `IcebergTableProperties` instead? This seems to be a more idiomatic way of doing this now that I looked here again

Reply:
Spent some time on this with @pajaks today. A better-informed remark on this:
- since we already have `compression-codec` as a session property, it would make it difficult to support it in `IcebergTableProperties` as a full-fledged table property (meaning: adding it to `ALTER TABLE table_name SET PROPERTIES compression-codec = 'xxx';` - no other Iceberg table property has a corresponding session property and it is [for a reason](https://github.com/trinodb/trino/blob/62f7a0bf21e1a4ff8e2d336a4fdc255cb7ab4d5c/core/trino-main/src/main/java/io/trino/execution/CreateTableTask.java#L159), although I don't think I grasp it in its entirety just yet